### PR TITLE
[feature/semantic-text] Add query and model compatibility tests

### PR DESCRIPTION
**Pull request by carlosdelest**
Add queries that showcase using nested queries for performing search without the need for `semantic_query`.

### CODE REVIEW

File: x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml [hunk 1]
```diff
@@ -50,19 +51,109 @@ setup:
...
+                    field: dense_field.inference.chunks.embeddings
+                    query_vector_builder:
+                      text_embedding:
+                        model_id: dense-inference-id
+                        model_text: "you know, for testing"
+
+    - match: { hits.total.value: 2 }
+    - match: { hits.total.relation: eq }
+    - match: { hits.hits.0.inner_hits.dense_field\.inference\.chunks.hits.total.value: 2 }
+    - exists:  hits.hits.0.inner_hits.dense_field\.inference\.chunks.hits.hits.0.fields.dense_field\.inference\.chunks.0.text
```
Comment (jimczi) [inline x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml:106]: We should be able to match the exact text of the chunk?

File: x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml [hunk 2]
```diff
@@ -50,19 +51,109 @@ setup:
...
+                query:
+                  text_expansion:
+                    sparse_field.inference.chunks.embeddings:
+                      model_id: sparse-inference-id
+                      model_text: "you know, for testing"
+
+    - match: { hits.total.value: 2 }
+    - match: { hits.total.relation: eq }
+    - match: { hits.hits.0.inner_hits.sparse_field\.inference\.chunks.hits.total.value: 2 }
+    - exists:  hits.hits.0.inner_hits.sparse_field\.inference\.chunks.hits.hits.0.fields.sparse_field\.inference\.chunks.0.text
```
Comment (jimczi) [inline x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml:156]: Same here?

**Review by jimczi — APPROVED** [review]
Thanks @carlosdelest

File: x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml (hunk 1 above)
Comment (carlosdelest) [reply x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml:106]: Sorry @jimczi  - merged already. But I've added this as part of #107262 now 👍

**Pull request merged**